asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+# <p>
+# N.B. Merging two Select all that apply type crosstab dataframes
+# <p>
+# Update: Free text responses recoded for df_605_crosstab (SQ44a, 44b) on 2024-04-17
+
+# %%
+# Display dataframes to be merged
+df_605_crosstab
+df_614_crosstab
+
+# %%
+# Create new df_614_crosstab with dummy column to match df_605_crosstab
+df_614_crosstab_mapped = df_614_crosstab.copy()
+df_614_crosstab_mapped["Only work with certain heat pump models"] = 0
+
+# Merge dataframes
+# N.B. Need to rename header of columns - SQ44 should be RQ6.16
+df_616_crosstab = df_605_crosstab.add(df_614_crosstab_mapped)
+
+# Save to .csv and display
+df_616_crosstab.to_csv("../../outputs/section6/csv/recoded/616_recoded.csv")
+df_616_crosstab
+
+# %%
+# Generate figure
+groupedbar(df_616_crosstab,
+           "616",
+           "RQ 6.16 Why do those who only offer, or only intend to offer, repairs, servicing and maintenance to heat pump systems they’ve installed choose not to do so for other systems?")
+
+# %%
+# Generate figure
+donut(df_616_crosstab,
+           "616",
+           "RQ 6.16 Why do those who only offer, or only intend to offer, repairs, servicing and maintenance to heat pump systems they’ve installed choose not to do so for other systems?")
+
+# %% [markdown]
+# ### RQ 6.17 | How do the reasons that survey participants give for only offering, or only intending to offer, repairs, servicing and maintenance to heat pump systems they’ve installed (but not for all other systems) differ depending on the size of the company they work for or own?
+# - SQ: 49a, 49b (intend to offer); 44a, 44b (offer)
+# - Sample:
+# - Company_size_owner
+# - Company size_employee
+# <p>
+# N.B. Merging two Select all that apply type crosstab dataframes
+# <p>
+# Update: Free text responses recoded for df_605_crosstab (SQ44a, 44b) on 2024-04-17
+
+# %%
+# CompanySizeOwner V2
+# Display dataframes to be merged
+df_606_crosstab_owner_V2
+df_615_crosstab_owner_V2
+
+# %%
+# CompanySizeOwner V2
+# Create new df_615_crosstab_owner_V2 with dummy column to match df_606_crosstab_owner_V2
+df_615_crosstab_owner_V2_mapped = df_615_crosstab_owner_V2.copy()
+df_615_crosstab_owner_V2_mapped["Only work with certain heat pump models"] = 0
+
+# Merge dataframes
+# N.B. Need to rename header of columns - SQ44 should be RQ6.17
+df_617_crosstab_owner_V2 = df_606_crosstab_owner_V2.add(df_615_crosstab_owner_V2_mapped)
+
+# Save to .csv and display
+df_617_crosstab_owner_V2.to_csv("../../outputs/section6/csv/recoded/617_owner_V2_recoded.csv")
+df_617_crosstab_owner_V2
+
+# %%
+# Generate figure
+groupedbar(df_617_crosstab_owner_V2,
+           "617_owner_V2",
+           "RQ 6.17 How do the reasons that survey participants give for only offering, or only intending to offer, repairs, servicing and maintenance to heat pump systems they’ve installed (but not for all other systems) differ depending on the size of the company they work for or own?")
+
+# %%
+# Generate figure
+donut(df_617_crosstab_owner_V2,
```

Comment:
Guess having a donut plot here doesn't matter too much since we don't have the granular level detail on the owner.